src/webhook_payload_template.js

```diff
@@ -0,0 +1,164 @@
+const { JsonToken, jsonStringifyExtended } = require('./utilities.client');
+
+/**
+ * WebhookPayloadTemplate enables creation and parsing of webhook payload template strings.
+ * Template strings are JSON that may include template variables enclosed in double
+ * curly brackets: `{{variable}}`. When the template is parsed, variables are replaced
+ * with values from a provided context.
+ *
+ * This is useful to create dynamic webhook payloads where a template is saved on webhook
+ * creation and then variables are dynamically added on webhook dispatch.
+ *
+ * **Example:**
+ * ```js
+ * const payloadTemplate = `{
+ *    "id": "some-id",
+ *    "createdAt": "2019-05-08T15:22:21.095Z",
+ *    "dataToSend": {{data}}
```

Comment:
From the comments it's not clear how to handle hierarchical objects, is it even possible? E.g. to use `{{data.status}}`

Reply:
It's not possible and it's in the OP of the PR.